Libraries/Components/DatePicker/DatePickerIOS.ios.js

```diff
@@ -122,7 +122,7 @@ var DatePickerIOS = React.createClass({
         <RCTDatePickerIOS
           ref={ picker => this._picker = picker }
           style={styles.datePickerIOS}
-          date={props.date.getTime()}
+          date={props.date}
```

Comment:
This only works by coincidence, because the JSONSerialization calls toString() on the Date object, and the native side a happens to support both strings and numbers. It's not a good fix though as it's relying on undocumented behavior.